test/extensions/filters/http/wasm/wasm_filter_integration_test.cc

```diff
@@ -234,6 +237,49 @@ TEST_P(WasmFilterIntegrationTest, BodyBufferedMultipleChunksManipulation) {
           "upstream_very_long_body.end");
 }
 
+TEST_P(WasmFilterIntegrationTest, LargeRequestHitBufferLimit) {
```

Comment:
I have tried to make the test failed, but with and without the change, both pass.

Reply:
this test mainly used to ensure big body could works as expected and this patch doesn't break anything.

It's hard to reproduce the bug in integration test because it requires specific language sdk implementation and the data should be handled by chunks.

I think I should add a unit test for this.

Reply:
The decode phase should not reproduce, as it will go through [prepareLocalReplyViaFilterChain](https://github.com/envoyproxy/envoy/blob/eb2033e982e5190785ab75a388b8383fd0cedbd5/source/common/http/filter_manager.cc#L1006-L1009) and will not immediately end the request

Reply:
> It's hard to reproduce the bug in integration test because it requires specific language sdk implementation and the data should be handled by chunks.

I didn't quite follow how the bug is SDK-dependent, I thought both Rust and C++ SDKs can return `StopIterationAndBuffer` (i.e. the value 1) from body callbacks. Or is it something else?